artemis-protocols/artemis-amqp-protocol/src/main/java/org/apache/activemq/artemis/protocol/amqp/proton/handler/ProtonHandler.java

```diff
@@ -213,9 +224,7 @@ public boolean isHandler() {
    }
 
    public void requireHandler() {
-      if (!workerExecutor.inEventLoop()) {
-         throw new IllegalStateException("this method requires to be called within the handler, use the executor");
-      }
+      assert workerExecutor.inEventLoop();
```

Comment:
Is the trade-off worth it? If it goes wrong in live usage then previously we would always find out. Now we probably wont, except any weird consequences of illegal use.

Also feels like a separate change either way.

Reply:
I placed it back...